table_engine/src/partition/mod.rs

```diff
@@ -101,42 +125,106 @@ impl From<meta_pb::Definition> for Definition {
-            PartitionInfo::Key(v) => Self::KeyPartition(meta_pb::KeyPartitionInfo {
-                definitions: v.definitions.into_iter().map(|v| v.into()).collect(),
-                partition_key: v.partition_key,
-                linear: v.linear,
-            }),
+            PartitionInfo::Hash(v) => meta_pb::PartitionInfo {
+                partition_info_enum: Some(meta_pb::partition_info::PartitionInfoEnum::Hash(
+                    meta_pb::HashPartitionInfo {
+                        definitions: v.definitions.into_iter().map(|v| v.into()).collect(),
+                        expr: Bytes::into(v.expr),
+                        linear: v.linear,
+                    },
+                )),
+            },
+            PartitionInfo::Key(v) => meta_pb::PartitionInfo {
+                partition_info_enum: Some(meta_pb::partition_info::PartitionInfoEnum::Key(
+                    meta_pb::KeyPartitionInfo {
+                        definitions: v.definitions.into_iter().map(|v| v.into()).collect(),
+                        partition_key: v.partition_key,
+                        linear: v.linear,
+                    },
+                )),
+            },
         }
     }
 }
 
-impl From<meta_pb::add_table_meta::PartitionInfo> for PartitionInfo {
-    fn from(pb: meta_pb::add_table_meta::PartitionInfo) -> Self {
-        match pb {
-            meta_pb::add_table_meta::PartitionInfo::Hash(v) => Self::Hash(HashPartitionInfo {
-                definitions: v.definitions.into_iter().map(|v| v.into()).collect(),
-                expr: Bytes::from(v.expr),
-                linear: v.linear,
-            }),
-            meta_pb::add_table_meta::PartitionInfo::KeyPartition(v) => {
-                Self::Key(KeyPartitionInfo {
+impl From<meta_pb::PartitionInfo> for PartitionInfo {
+    fn from(partition_info_pb: meta_pb::PartitionInfo) -> Self {
+        match partition_info_pb.partition_info_enum {
+            Some(partition_info_enum) => match partition_info_enum {
+                meta_pb::partition_info::PartitionInfoEnum::Hash(v) => {
+                    Self::Hash(HashPartitionInfo {
+                        definitions: v.definitions.into_iter().map(|v| v.into()).collect(),
+                        expr: Bytes::from(v.expr),
+                        linear: v.linear,
+                    })
+                }
+                meta_pb::partition_info::PartitionInfoEnum::Key(v) => Self::Key(KeyPartitionInfo {
                     definitions: v.definitions.into_iter().map(|v| v.into()).collect(),
                     partition_key: v.partition_key,
                     linear: v.linear,
-                })
-            }
+                }),
+            },
+            // TODO: How to return empty enum ? This return value needed to be fixed.
+            None => PartitionInfo::Key(KeyPartitionInfo {
+                definitions: vec![],
+                partition_key: vec![],
+                linear: false,
+            }),
```

Comment:
`From` trait here may be not suitable, we may need `TryFrom`.
When encounter `None`, we should return error.